packages/utils/src/SDKErrors.ts

```diff
@@ -6,348 +6,225 @@
  */
 
 /**
- * SDKErrors are KILT-specific errors, with associated codes and descriptions.
+ * KILT-specific errors with descriptions.
  *
  * @packageDocumentation
  */
 
 /* eslint-disable max-classes-per-file */
 
-export abstract class SDKError extends Error {
-  constructor(...args: ConstructorParameters<ErrorConstructor>) {
-    super(...args)
+export class SDKError extends Error {
+  constructor(message?: string) {
```

Comment:
There are potentially more arguments on the Error class, which we want to use in the future, e.g. `options.cause`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error

Reply:
Yeah, too bad the current version of TypeScript doesn’t allow them. Could we add them when this will be possible? The `...args` type seems a bit clumsy?

Reply:
With this commit it is now supported ;) b57c174ddbe63ef83ab92ea9dc658a21b62174fe

Reply:
I’ve updated the PR and also included causes in the places where we wrap errors